mmv1/templates/terraform/examples/alloydb_backup_basic.tf.erb

```diff
@@ -16,6 +16,12 @@ resource "google_alloydb_instance" "<%= ctx[:primary_resource_id] %>" {
   cluster       = google_alloydb_cluster.<%= ctx[:primary_resource_id] %>.name
   instance_id   = "<%= ctx[:vars]['alloydb_instance_name'] %>"
   instance_type = "PRIMARY"
+  client_connection_config {
```

Comment:
Does this block need to be specified in every example?

Reply:
Technically no. If the user doesn't specify it, instance creation will still work as expected. The problem is that `TestAccAlloydbInstance_alloydbInstanceBasicExample` test will fail without it because in the pre_create code, we manually create a `client_connection_config` for the instance even if the user didn't have it in their terraform file. Terraform seems to get confused and expects a nil value for `client_connection_config` then for the plan - even though we don't want the value to be nil. 

If there's a better way to do this so we can remove it here, then do let me know! I think I tried to use `allow_empty_object` but I remember that didn't exactly work for being able to set the default SSL mode

Reply:
What happens if you don't use the pre create but instead set `default_from_api` on the field?

Reply:
We can't use `default_from_api` because we haven't yet actually set the default to be `ENCRYPTED_ONLY` on the API side for the SSL mode. This is because there's a prerequisite task that needs to be done first before we do so. You can look at b/293535381#comment7 and b/300480358 for the actual prereq task. 

Once we actually have set the default on the API side (and that change has reached prod), I plan on removing the custom code, updating the tests, and just use `default_from_api` instead.

Reply:
Also, I didn't want to use `default_value` temporarily until we set the default on the API side since if we were to change the default value in the future then it would be a breaking change. I'm not sure if changing it from `default_value` to `default_from_api` however is a breaking change per say if the underlying value is the same. Though I don't anticipate we will change the default, I think it's better to use the custom code instead for now.